variables.tf

```diff
@@ -304,6 +304,10 @@ variable "bitbucket_display_name" {
   description = "The display name of Bitbucket instance"
   type        = string
   default     = null
+  validation {
```

Comment:
One validation is sufficient. Because we not suggesting to use the modules individually, having one validation in root level would be sufficient and visible. The maintenance would be easier if we need to change the validations in future as well.

Reply:
Terraform question - if this validation is at this level does it mean invalid strings caught and flagged sooner?

Reply:
not really, because validation will happening in planning stage. And terraform will recursively look for all modules to complete the plan. But is good to put the validations in root level to be more visible to users. They just need to look at the root variables.tf to see the validations, and they will not hide in module structures. 
Having the validation in module level is very good when modules can be re-use individually, but in our project, we do not intend to produce the modules to be used as a single unit.

Reply:
yup, fair enough~ But instead of root module, I'd like to keep it in Bitbucket module.

Reply:
@yzha645 , following our chat, you pointed a good point that we use the validation in unit test, so it would be fine to have it in module level, but please remove the redundant validation code from root level. Repeating the same code in two files makes us a difficult life in maintaining the code.